### DISCUSSION TITLE
Can't find a good block size

### DISCUSSION THREAD

**Discussion by lionelalexandre** [OrderN/CONQUEST-release#112] [Running]
Dear All,

I've got an issue in running Conquest in parallel with diagonalisation ; I get "Can't find a good block size...".
I've looked quickly to this part of the code and docs and I think this is related to the fact that the H matrix size
is a prime number (unlucky !). Is there a way of dealing with that ? 

Thank you for your precious help.
Lionel

**Comment by davidbowler** [comment]
The only solution at the moment is to set the block size to one using `Diag.BlockSizeR` and `Diag.BlockSizeC`.  This may make the diagonalisation less efficient.

We do have a possible fix which involves padding the Hamiltonian to make it a good size and then discarding the resulting eigenstates, but that is not available yet (though soon!).
